src/org/parosproxy/paros/network/SSLConnector.java

```diff
@@ -539,7 +541,7 @@ public Socket createTunnelServerSocket(String targethost, Socket socket) throws
 	}
 
 	// ZAP: added new ServerSocketFaktory with support of dynamic SSL certificates
-	public SSLSocketFactory getTunnelSSLSocketFactory(String hostname) {
+	public SSLSocketFactory getTunnelSSLSocketFactory(String hostname, InetAddress listeningAddress) {

```

Comment:
In this case we should also introduce a new method (and deprecate the "old" one?).